test-tools/src/test/java/org/terracotta/utilities/test/matchers/EventuallyTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Terracotta, Inc., a Software AG company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.utilities.test.matchers;
+
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.any;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.stringContainsInOrder;
+import static org.junit.internal.matchers.ThrowableMessageMatcher.hasMessage;
+import static org.terracotta.utilities.test.matchers.Eventually.within;
+import static org.terracotta.utilities.test.matchers.ThrowsMatcher.threw;
+
+public class EventuallyTest {
+
+  @Test
+  public void testImmediateTrue() {
+    assertThat(() -> true, within(Duration.ZERO).is(true));
```

Comment:
Same comment  - this is not really hamcrest-like with the .is(), .matches, etc.
Could we instead have something more hamcrest like ?
=> perhaps: `assertThat(() -> true, within(Duration.ZERO, is(true)));` ?